src/object/srv_obj.c

```diff
@@ -687,6 +687,133 @@ obj_echo_rw(crt_rpc_t *rpc, daos_iod_t *split_iods, uint64_t *split_offs)
 	orwo->orw_map_version = orw->orw_map_ver;
 }
 
+
+int
+ec_bulk_transfer(crt_rpc_t *rpc, crt_bulk_op_t bulk_op, bool bulk_bind,
+		 crt_bulk_t *remote_bulks, daos_handle_t ioh,
+		 struct ec_bulk_spec **skip_list,
+		 int sgl_nr)
+{
+	struct obj_bulk_args	arg = { 0 };
+	crt_bulk_opid_t		bulk_opid;
+	crt_bulk_perm_t		bulk_perm = CRT_BULK_RW;
+	int			i, rc, *status, ret;
+
+	D_ASSERT(skip_list != NULL);
+
+	rc = ABT_eventual_create(sizeof(*status), &arg.eventual);
+	if (rc != 0)
+		return dss_abterr2der(rc);
+
+	arg.bulks_inflight++;
+	for (i = 0; i < sgl_nr; i++) {
+		d_sg_list_t		*sgl, tmp_sgl;
+		struct crt_bulk_desc	 bulk_desc;
+		crt_bulk_t		 local_bulk_hdl;
+		struct bio_sglist	*bsgl;
+		daos_size_t		 offset = 0;
+		unsigned int		 idx = 0;
+		unsigned int		 sl_idx = 0;
+
+		if (remote_bulks[i] == NULL)
+			continue;
+
+		D_ASSERT(!daos_handle_is_inval(ioh));
```

Comment:
Loop invariant so could be moved to L703.

Reply:
done